q2.java

```diff
@@ -0,0 +1,113 @@
+
+
+//Implementing a linked list
+class LinkedList<T>{
+	Node<T> head;
+	
+	public LinkedList(){
+		head = null;
+	}
+	
+	public void appendLast(T data){
+		Node<T> last = new Node<T>(data);
+		Node<T> current = this.head;
+		if(current == null){
+			head = last;
+			return;
+		}
+		while(current.next!=null){
+			current = current.next;
+		}
+		current.next=last;		
+	}
+	
+	/** 
+	 * The idea is to create a window of length k where
+	 * right margin is pointed by current
+	 * left margin is pointed by follower
+	 * when current reaches end of linked linked, follower points to kth to last element
+	 */
+	public Node<T> kToLast(int k){
+		if(k<0){
+			System.err.println("k must be a non-negative number");
+			return null;
+		}
+		
+		Node<T> current = head;
+		for(int i=0; i<k; i++){
+			if(current==null){
+				break; // this condition holds if k>=list.length+1
+			}
+			current = current.next;
+		}
+		if(current==null){
+			return null; //this condition holds if k>=list.length 
+		}
+		
+		Node<T> follower = head;
+		while(current.next!=null){
+			follower = follower.next;
+			current = current.next;
+		}
+		return follower;
+	}
+	
+	public void printList (){

```

Comment:
Overriding the toString() method would make more sense I think as it keeps the method of output separated from the creation of the output. One user may want to write to System.out; while another user may want to add the list contents to an error message.